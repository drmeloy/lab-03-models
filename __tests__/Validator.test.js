const Validator = require('../lib/Validator');

describe('Age validator', () => {
  let validator;

  describe('when field is required', () => {
    beforeAll(() => {
      validator = new Validator('age', {
        type: Number,
        required: true
      });
    });

    it('returns the field\'s value', () => {
      const cat = {
        name: 'Meowzors',
        age: 7,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual(7);
    });

    it('returns the field\'s value cast to proper type', () => {
      const cat = {
        name: 'Meowzors',
        age: '8',
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual(8);
    });

    it('returns that the field is required if no value given', () => {
      const cat = {
        name: 'Meowzors',
        weight: '6 lbs'
      };

      expect(() => validator.validate(cat)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when field is not required', () => {
    beforeAll(() => {
      validator = new Validator('age', {
        type: Number
      });
    });

    it('returns the field when given a value', () => {
      const cat = {
        name: 'Meowzors',
        age: 6,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual(6);
    });

    it('returns the field cast to proper type', () => {
      const cat = {
        name: 'Meowzors',
        age: '6',
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual(6);
    });

    it('returns null when field is given no value', () => {
      const cat = {
        name: 'Meowzors',
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toBeNull();
    });

  });  
});

describe('Name validator', () => {
  let validator;

  describe('when field is required', () => {
    beforeAll(() => {
      validator = new Validator('name', {
        type: String,
        required: true
      });
    });

    it('returns the field\'s value', () => {
      const cat = {
        name: 'Meowzors',
        age: 7,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual('Meowzors');
    });

    it('returns the field\'s value cast to proper type', () => {
      const cat = {
        name: 88,
        age: 8,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual('88');
    });

    it('returns that the field is required if no value given', () => {
      const cat = {
        age: 8,
        weight: '6 lbs'
      };

      expect(() => validator.validate(cat)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('when field is not required', () => {
    beforeAll(() => {
      validator = new Validator('name', {
        type: String
      });
    });

    it('returns the field when given a value', () => {
      const cat = {
        name: 'Meowzors',
        age: 6,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual('Meowzors');
    });

    it('returns the field cast to proper type', () => {
      const cat = {
        name: 88,
        age: 6,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toEqual('88');
    });

    it('returns null when field is given no value', () => {
      const cat = {
        age: 6,
        weight: '6 lbs'
      };

      expect(validator.validate(cat)).toBeNull();
    });

  });  
});
