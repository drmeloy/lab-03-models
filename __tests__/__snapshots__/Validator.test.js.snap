// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Age validator when field is required returns that the field is required if no value given 1`] = `"age is required"`;

exports[`Name validator when field is required returns that the field is required if no value given 1`] = `"name is required"`;
